### ISSUE TITLE
Add more Objects manipulation functions to Container, and deprecate direct slice access

### ISSUE THREAD

**Issue by dweymouth** [fyne-io/fyne#4599]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

As an ongoing effort for thread-safety, user code should be discouraged from manipulating exported fields directly, especially slices. We should provide several more functions to manipulate the Objects slice in Container in a thread-safe way, such as:

* container.Replace(objA, objB)
* container.ReplaceAt(int, obj)
* container.RemoveAt(int)
* container.Swap(int, int)
* container.Insert(int, obj)
* container.SetObjects(objects) - for whole content replacement

etc...

Basically make it possible to do everything needed without directly manipulating the Objects slice, and then deprecate access to the Objects slice directly.

### Describe the solution you'd like to see.

described above.

**Comment by dweymouth** [comment]
This won't be necessary with the new proposed threading model.

**Comment by andydotxyz** [comment]
Won't some of these like SetObjects still be helpful?

**Comment by dweymouth** [comment]
Sure, I guess so. I suppose it depends if we want to standardize that Setters are thread-safe (and internally `wid.SetFoo(fooVal)` is basically `fyne.RunOnMain(func() { wid.Foo = fooVal; wid.Refresh() })`

If we go this way, then yes `container.SetObjects` could just be runOnMain ( () => container.Objects = obj )

**Comment by dweymouth** [comment]
Without the new threading model, `container.SetObjects` would be absolutely needed, as in there's no other safe way to accomplish it. With the new thread model, it would be a shorthand

**Comment by andydotxyz** [comment]
I think where we got to on the call is that Setters are guaranteed thread safe, as was intended in current code. It may just be a shorthand but it's an intuitive one.
